### PR TITLE
[WIP] integrate keystone-storage-adapter-s3

### PR DESCRIPTION
**Pull request by radiovisual**
This PR is a Work in Progress

At this point, the PR fails to run, due to Keystone error: 

<pre>
❯ node keystone           
/Users/michael/forks/keystone/fields/types/file/FileType.js:33
    Object.keys(this.storage.schema).forEach(function (path) {
           ^

TypeError: Cannot convert undefined or null to object
    at Function.keys (native)
    at file.addToSchema (/Users/michael/forks/keystone/fields/types/file/FileType.js:33:9)
    at file.Field (/Users/michael/forks/keystone/fields/types/Type.js:82:7)
    at new file (/Users/michael/forks/keystone/fields/types/file/FileType.js:19:14)
    at List.field (/Users/michael/forks/keystone/lib/list/field.js:45:14)
    at List.<anonymous> (/Users/michael/forks/keystone/lib/list/add.js:41:16)
    at List.<anonymous> (/Users/michael/forks/keystone/lib/list/add.js:30:5)
    at /Users/michael/forks/keystone/lib/list/add.js:74:5
    at arrayEach (/Users/michael/forks/keystone/node_modules/lodash/lodash.js:484:11)
    at Function.forEach (/Users/michael/forks/keystone/node_modules/lodash/lodash.js:8843:14)
    at List.add (/Users/michael/forks/keystone/lib/list/add.js:48:4)
    at Object.<anonymous> (/Users/michael/forks/keystone-test-project/models/Post.js:11:6)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
</pre>

### CODE REVIEW

**Comment by radiovisual** [comment]
I have been experimenting, but haven't been able to get the S3 adapter to play nicely yet due to a problem with reading the storage schema. For now, I am just using the `Post.js` in my functional tests, trying to get the image in the Post to upload to S3.

File: models/File.js [hunk 1]
```diff
@@ -8,20 +8,36 @@ var storage = new keystone.Storage({
 	adapter: keystone.Storage.Adapters.FS,
 	fs: {
 		path: keystone.expandPath('./uploads'),
-		publicPath: '/public/uploads/',
+		publicPath: '/public/uploads/'
```
Comment (JedWatson) [inline models/File.js:11]: Heads up - our linter checks for trailing commas

File: models/Post.js [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
 });
 
+
 Post.add({
 	name: { type: String, required: true },
 	state: { type: Types.Select, options: 'draft, published, archived', default: 'draft', index: true },
 	author: { type: Types.Relationship, ref: 'User', index: true },
 	publishedDate: { type: Types.Date, index: true, dependsOn: { state: 'published' } },
-	image: { type: Types.CloudinaryImage },
+	image: { type: Types.File, storage: 'S3Storage' },
```
Comment (JedWatson) [inline models/Post.js:16]: This is probably why keystone is failing to start - storage needs to be a storage instance, it's a string here

**Comment by JedWatson** [comment]
@radiovisual I've added this with the right config, am still seeing issues with the Adapters but I'm working in those repos directly and keystone to debug. Will keep you posted with progress over in #16

**Pull request closed without merging**